src/euphorie/client/browser/report.py

```diff
@@ -252,4 +249,9 @@ def __call__(self):
             "Content-Type",
             "application/vnd.openxmlformats-" "officedocument.spreadsheetml.sheet",
         )
-        return save_virtual_workbook(book)
+
+        with NamedTemporaryFile(delete=False) as tmp:
```

Comment:
It seems strange to me that we have delete False.

Reply:
It's modelled after this example: https://docs.python.org/3/library/tempfile.html#examples

Note that on python 3.13 the parameter is `delete_on_close` which is a more accurate name.

Reply:
The context manager is doing that. The parameter only prevents deletion when `close()` is called. When the context is exited the file is deleted anyway.

The documentation is a bit ambiguous, but that's how I understand it.

> If _delete_ is true (the default), the file is deleted as soon as it is closed.

https://docs.python.org/3.11/library/tempfile.html#tempfile.NamedTemporaryFile

In 3.13 there is an improvement:

> If _delete_ is true and _delete_on_close_ is false, the file is deleted on context manager exit only

https://docs.python.org/3/library/tempfile.html#tempfile.NamedTemporaryFile

Reply:
Meh... I tested it - the file is not cleaned up by the context manager. I'll change the code.